### PR TITLE
Add NodeOption, ServiceClientOption, and ServiceServerOption

### PR DESCRIPTION
**Pull request by aeneev**
These options allow to customize created node instances.
This commit adds the first options that allow to customize tcp timeouts
of service clients and servers.

This is helpful for automated tests to reduce test flakiness driven by
varying network and CPU performance on the test machines.

### CODE REVIEW

**Comment by aeneev** [comment]
@cjds thank you for reviewing the other two PRs.

This PR is more of a feature than a bugfix, but it did help us resolve a lot of flakiness for the tests of the system I've integrated rosgo into.

Please let me know if you have any comments or if someone else should review this.

Thank you,
Alan

**Comment by cjds** [comment]
I'll review it. I do want to run this through our test suite before merging it. I'm going to work on integrating that first. I don't think I have an issue with any of it but might hold off on merging for a day or 2

**Comment by aeneev** [comment]
No rush on merging this in - I just wanted to make sure it is on someones radar. Thanks for looking into it.

**Review by cjds — APPROVED** [review]
Tested locally LGTM

**Pull request merged**
